src/components/TldrawCanvasLabelDialog.tsx

```diff
@@ -306,15 +308,19 @@ const LabelDialog = ({
   const containerRef = useRef<HTMLDivElement>(null);
   const [error, setError] = useState("");
   const initialLabel = useMemo(() => {
-    if (_label) return _label;
+    if (_label)
+      return getPageTitleByPageUid(initialUid) || getTextByBlockUid(initialUid);
     const { specification, text } = discourseContext.nodes[nodeType];
     if (!specification.length) return "";
     return getPlainTitleFromSpecification({ specification, text });
-  }, [_label, nodeType]);
+  }, [_label, nodeType, initialUid, isOpen]);
   const initialValue = useMemo(() => {
     return { text: initialLabel, uid: initialUid };
   }, [initialLabel, initialUid]);
-  const [label, setLabel] = useState(initialValue.text);
+  const [label, setLabel] = useState("");
+  useEffect(() => {
+    setLabel(initialLabel);
```

Comment:
I think bc `_label` updates, `initialLabel` updates, so doesn't this fire on every keystroke?

Reply:
I think `_label` only updated when `shape.props.title` changed, which is `onSuccess` of the dialog.

https://www.loom.com/share/09f71b10c61b4dbaa8aad2bbb61f0308

Regardless, testing this showed that it was firing for every node on canvas load.  Changed it to fire on `isOpen`